Handle repeated field lookups in calls to values_list

leamingrad opened the pull request:
Prior to this change, if checked code was calling the values_list method
on a queryset with a repeated field lookup, for example
Model.objects.values_list("foo", "foo"), then we would fail to type the
resulting tuple correctly. The second argument would be untyped, and the
resulting tuple would have fewer fields than Django would actually give
it.

This problem only affects unnamed, non-flat tuples, as Django will error
if a repeating field lookups is attempted for these other variants.

This change ensures that when we type an unnamed tuple, we refer back to
the original field lookups when passing in the resolved values. We also
add a test to explain the issue, and prevent regressions.


## Related issues

I could not find a related issue for this, but I can raise one if that
would be helpful.

sobolevn commented on tests/typecheck/managers/querysets/test_values_list.yml:
```diff
@@ -26,6 +26,25 @@
+            content: |
+                from django.db import models
+                class MyUser(models.Model):
+                    name = models.CharField(max_length=100)
```
Can we please have another field for this test? Something like `age = models.IntegerField`

leamingrad replied:
Done - good suggestion!

sobolevn commented on tests/typecheck/managers/querysets/test_values_list.yml:
```diff
@@ -26,6 +26,25 @@
+
+        # Values tuples can have the same field repeated
+        values_tuple = MyUser.objects.values_list('name', 'name').get()
+        reveal_type(values_tuple)  # N: Revealed type is "Tuple[builtins.str, builtins.str]"
```
And let's use `age` here as well to be sure that this is not just a single type that repeats itself :)

leamingrad replied:
Done - good suggestion!. I've doubled-checked that without the fix the new version of the test does fail.

sobolevn reviewed (commented):
Thanks, just one proposal :)

leamingrad committed:
Handle repeated field lookups in calls to values_list

Prior to this change, if checked code was calling the values_list method
on a queryset with a repeated field lookup, for example
Model.objects.values_list("foo", "foo"), then we would fail to type the
resulting tuple correctly. The second argument would be untyped, and the
resulting tuple would have fewer fields than Django would actually give
it.

This problem only affects unnamed, non-flat tuples, as Django will error
if a repeating field lookups is attempted for these other variants.

This change ensures that when we type an unnamed tuple, we refer back to
the original field lookups when passing in the resolved values. We also
add a test to explain the issue, and prevent regressions.

sobolevn reviewed (approved):
Great job! 👍
